crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -4160,6 +4165,76 @@ impl<'db> TypeInferenceBuilder<'db> {
         Ok(builder.build())
     }
 
+    fn try_infer_type_api_operation(
+        &mut self,
+        api_type: Type<'db>,
+        arguments: &ast::Expr,
+    ) -> Option<Type<'db>> {
+        match api_type.into_class_literal() {
+            Some(class)
+                if file_to_module(self.db(), class.class.file(self.db()))
+                    .is_some_and(|module| module.is_known(KnownModule::RedKnot)) =>
+            {
+                let db = self.db();
+
+                let is_type_of = class.class.name(db).as_str() == "TypeOf";
+
+                let argument_types = match arguments {
+                    ast::Expr::Tuple(tuple) => Either::Left(
+                        tuple
+                            .elts
+                            .iter()
+                            .map(|element| self.infer_type_expression(element)),
```

Comment:
nit: you can iterate directly over the tuple:

```suggestion
                        tuple
                            .iter()
                            .map(|element| self.infer_type_expression(element)),
```